src/world.js

```diff
@@ -80,13 +83,11 @@ export default class World {
         this.addNextBlock();
       }
     }
-
-    // if any clouds off screen, remove them
   }
 
   updateClouds() {
     this.clouds.forEach((cloud, index) => {
```

Comment:
We could re-write this a bit, and do something like:

```js
const pLength = this.clouds.length;
this.clouds = this.clouds.filter(c => (c.x + c.width) >= 0);
if (this.clouds.length !== pLength) {
  this.addNextCloud();
}
```
Both splice and addNextCloud are mutating functions, which makes the `forEach` seem unpredictable.

Reply:
Yeah I like that rewrite - I wondered whether those two functions would at some point interact in a hilarious way. It sadly hasn't obliged with funny stuff, yet, though :D

Reply:
Also, we iterate through each cloud every time - we have around 9 clouds so it's not too bad - but how do you feel about some comparator / filter function that we can pass to the position update? Then it can remove the off-screen ones as it encounters them - or maybe build up a list of off-screen ones that can be passed to a removal function, or some such?

Reply:
Maybe not in *this* PR, though - but yeah throwing around some ideas about what gets passed to the rendering pipeline

Reply:
I guess my only thoughts on that are the haunting, echo-y voices saying "Separation of Concerns!" :laughing: . That said, let's add an issue on the board for it, and we can start to figure out what that implementation would look like.